src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -881,13 +924,23 @@ public Builder withOkHttpClient(OkHttpClient okHttpClient) {
          * @return A new {@link Pinecone} instance configured based on the builder parameters.
          */
         public Pinecone build() {
-            PineconeConfig clientConfig = new PineconeConfig(apiKey);
-            clientConfig.setSourceTag(sourceTag);
-            clientConfig.validate();
+            PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig);
+            config.validate();
+
+            if(isProxyConfigured && isCustomOkHttpClient) {
+                throw new PineconeConfigurationException("Invalid configuration: Both Custom OkHttpClient and Proxy are set. Please configure only one of these options.");
+            }
+
+            OkHttpClient.Builder okHttpClientBuilder = new OkHttpClient.Builder();
+
+            if(isProxyConfigured) {
+                Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort()));
+                okHttpClientBuilder.proxy(proxy);
+            }
 
-            ApiClient apiClient = new ApiClient(okHttpClient);
-            apiClient.setApiKey(clientConfig.getApiKey());
-            apiClient.setUserAgent(clientConfig.getUserAgent());
+            ApiClient apiClient = (isCustomOkHttpClient) ? new ApiClient(customOkHttpClient) : new ApiClient(okHttpClientBuilder.build());
```

Comment:
This feels a little awkward to me, how it still creates the OkClient even if there is a custom client. Is this any better?

```java
        private OkHttpClient buildOkHttpClient() {
            OkHttpClient.Builder builder = new OkHttpClient.Builder();
            if(isProxyConfigured) {
                Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort()));
                builder.proxy(proxy);
            }
            return builder.build();
        }

        public Pinecone build() {
            PineconeConfig config = new PineconeConfig(apiKey, sourceTag, proxyConfig);
            config.validate();

            if(isProxyConfigured && isCustomOkHttpClient) {
                throw new PineconeConfigurationException("Invalid configuration: Both Custom OkHttpClient and Proxy are set. Please configure only one of these options.");
            }

            ApiClient apiClient = (isCustomOkHttpClient) ? new ApiClient(customOkHttpClient) : new ApiClient(buildOkHttpClient());
     
...........
```

Reply:
Yeah makes sense, so let me know what you think now. The only change is the name of the method and removing the check on if proxy is configured or not, since the proxy has to be configured if the method is called.
```java
        private OkHttpClient buildProxyConfiguredOkHttpClient() {
            OkHttpClient.Builder builder = new OkHttpClient.Builder();
            Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort()));
            builder.proxy(proxy);
            return builder.build();
        }
```

Reply:
yeah makes sense